components/ratgdo/ratgdo.cpp

```diff
@@ -232,11 +268,57 @@ namespace ratgdo {
         }
     }
 
+    void RATGDOComponent::setOpeningDuration(float duration)
+    {
+        ESP_LOGD(TAG, "Set opening duration: %.1fs", duration);
+        this->openingDuration = duration;
+        this->openingDurationPref_.save(&this->openingDuration);
+        set_timeout("sync_preferences", 500, [=] {
+            global_preferences->sync();
+        });
```

Comment:
I think its best to let the 5s duration elapse and have preferences write it here